src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -75,17 +75,6 @@ CARTree::~CARTree()
 {
 }
 
-void CARTree::set_labels(std::shared_ptr<Labels> lab)
-{
-	if (lab->get_label_type()==LT_MULTICLASS)
-		set_machine_problem_type(PT_MULTICLASS);
-	else if (lab->get_label_type()==LT_REGRESSION)
-		set_machine_problem_type(PT_REGRESSION);
```

Comment:
is this not required?

Reply:
how about move those code to `train_machine`.

Reply:
Not sure, I wonder if it is needed?